apps/andi/lib/andi_web/live/ingestion_live_view/edit_ingestion_live_view.ex

```diff
@@ -194,34 +256,80 @@ defmodule AndiWeb.IngestionLiveView.EditIngestionLiveView do
     {:noreply, redirect(socket, to: socket.assigns.unsaved_changes_link)}
   end
 
+  def handle_event(event, payload, socket) do
+    IO.inspect(__MODULE__, label: "Module")
+    IO.inspect(event, label: 'Unhandled Event in module #{__MODULE__}')
+    IO.inspect(payload, label: 'Unhandled Payload in module #{__MODULE__}')
+    IO.inspect(payload, label: 'Unhandled Socket in module #{__MODULE__}')
+
+    {:noreply, socket}
+  end
+
   def test_url(socket) do
     save_ingestion(socket)
   end
 
+  defp save_ingestion_safe(socket) do
+    # Once all subforms are routed through this parent live view, this save function
+    # can save directly to the Repo from socket.assigns.changeset without having to extract
+    # the changes and reapply to the ingestion from the database, but for now, we need to treat
+    # the database as the source of truth that can change at any time.
+
+    safe_ingestion_data = socket.assigns.changeset
+                          |> Changeset.apply_changes
+
+    safe_extracted_data = %{
+      name: safe_ingestion_data.name,
+      sourceFormat: safe_ingestion_data.sourceFormat,
+      targetDataset: safe_ingestion_data.targetDataset,
+      topLevelSelector: safe_ingestion_data.topLevelSelector
+    }
+
+    current_ingestion = Ingestions.get(socket.assigns.ingestion.id)
+
+    case Ingestions.update(current_ingestion, safe_extracted_data) do
+      {:ok, post_save_ingestion} ->
+        post_save_ingestion
+      {error, details} ->
+        raise "Unable to save ingestion. Error: #{error}. Details: #{details}"
+    end
+
+
+  end
+
   defp save_ingestion(socket) do
     ingestion_id = socket.assigns.ingestion.id
+
     AndiWeb.Endpoint.broadcast_from(self(), "form-save", "save-all", %{ingestion_id: ingestion_id})
     # Todo: Rearchitect how concurrent events are handled and remove these sleeps from draft-save and publish of datasets and ingestions
     # This sleep is needed because other save events are executing. publish_ingestion will load the ingestion from the database.
     Process.sleep(1_000)
 
     andi_ingestion = Ingestions.get(ingestion_id)
     ingestion_changeset = InputConverter.andi_ingestion_to_full_ui_changeset(andi_ingestion)
+    updated_socket = assign(socket, changeset: ingestion_changeset)
```

Comment:
The current socket is out-of-sync from the database at this point, so you must construct a new changeset from the up-to-date data from the database, assign it to the socket, then use that to determine the correct save message :(

Reply:
This is ":(", but a reminder to others reviewing that this is an existing issue, not new as part of this refactor 👍 

When other components switch over to livecomponent, and all utilize "safe_ingestion_save", this extra step will go away.

Prior to this PR, save messages were at times inaccurate, this should improve their accuracy by checking with the database.